src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriterCache.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines a thread-local cache for JsonSerializer to store reusable Utf8JsonWriter/IBufferWriter instances.
+    /// </summary>
+    internal static class Utf8JsonWriterCache
+    {
+        [ThreadStatic]
+        private static ThreadLocalState? s_threadLocalState;
```

Comment:
s_threadLocalState => t_threadLocalState

Reply:
Interesting, I wasn't even aware of that naming convention.